backend/src/main/java/gov/cdc/usds/simplereport/logging/RestAuditLogManager.java

```diff
@@ -0,0 +1,49 @@
+package gov.cdc.usds.simplereport.logging;
+
+import gov.cdc.usds.simplereport.api.pxp.CurrentPatientContextHolder;
+import gov.cdc.usds.simplereport.db.model.Organization;
+import gov.cdc.usds.simplereport.db.model.PatientLink;
+import gov.cdc.usds.simplereport.service.AuditService;
+import gov.cdc.usds.simplereport.service.errors.RestAuditFailureException;
+import javax.servlet.http.HttpServletRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+/** Wrapper around {@link AuditService} to catch exceptions and rethrow them */
+@Service
+// NOT TRANSACTIONAL (must catch errors at transaction boundaries)
+@SuppressWarnings("checkstyle:IllegalCatch")
+public class RestAuditLogManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RestAuditLogManager.class);
+
+  private final AuditService _auditService;
+  private final CurrentPatientContextHolder _contextHolder;
+
+  private static final int DEFAULT_SUCCESS = HttpStatus.OK.value();
+
+  public RestAuditLogManager(AuditService auditService, CurrentPatientContextHolder contextHolder) {
+    super();
+    this._auditService = auditService;
+    this._contextHolder = contextHolder;
+  }
+
+  public boolean logRestSuccess(HttpServletRequest request, Object resultObject) {
+    PatientLink patientLink = _contextHolder.getPatientLink();
+    if (patientLink == null) {
+      LOG.error("Somehow reached PXP success handler with no patient link");
+      return false;
```

Comment:
is there any impact on the annotated controller in the event of a false return here? I couldn't immediately find an answer in the spring-security docs, and it's certainly not in the javadoc

Reply:
It will convert the response to a 403, I believe. So yes. 😬